### PR TITLE
add support for search prop

### PR DESCRIPTION
**Pull request by jmisasa**
This PR adds support for [search prop](https://nodejs.org/api/url.html#urlsearch) which while using the library we noticed it was yet not supported

Thanks a lot for this amazing package!

### CODE REVIEW

**Comment by jmisasa** [comment]
I noticed this PR https://github.com/scottrippey/next-router-mock/pull/82 where node dependencies were eliminated. I just pushed some new changes that should address this. 

Didn't go with straight `URLSearchParams` because of the "bracket" notation required to parse a URL that contains multiple values for the same key

With [query-string](https://github.com/sindresorhus/query-string) this behaviour is allowed:

```ts
import queryString from 'query-string';

queryString.parse('foo=1&foo=2&foo=3');
//=> {foo: ['1', '2', '3']}
```

Happy to read your thoughts. Thank you!

**Comment by jmisasa** [comment]
ended up removing external dependency and used parsedUrl directly, feels kind of hacky but seems to do the trick. Thanks again!

File: src/urls.ts [hunk 1]
```diff
@@ -26,3 +26,8 @@ export function stringifyQueryString(query: NextRouter["query"]): string {
   });
   return params.toString();
 }
+export function objectifyQueryString(query: string): NextRouter["query"] | undefined {
```
Comment (scottrippey) [inline src/urls.ts:29]: This always returns a value, so:
```suggestion
export function objectifyQueryString(query: string): NextRouter["query"] {
```

File: src/MemoryRouter.test.tsx [hunk 2]
```diff
@@ -417,6 +417,55 @@ describe("MemoryRouter", () => {
...
+            expectMatch(memoryRouter, {
+              asPath: "/path?foo=FOO&foo=BAR",
+              pathname: "/path",
+              query: {
+                foo: ["FOO", "BAR"],
+              },
+            });
+          });
+
+          it("if search and query are both provided preference is given to search", async () => {
```
Comment (scottrippey) [inline src/MemoryRouter.test.tsx:450]: Not a big deal, but did you verify that this is how Next behaves?  Gives preference to `search` over `query`?

File: src/MemoryRouter.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import type { NextRouter, RouterEvent } from "next/router";
 import mitt, { MittEmitter } from "./lib/mitt";
-import { parseUrl, stringifyQueryString } from "./urls";
+import { parseUrl, objectifyQueryString, stringifyQueryString } from "./urls";
```
Comment (scottrippey) [inline src/MemoryRouter.tsx:3]: Could we rename this new method to `parseQueryString`?

**Review by scottrippey — CHANGES_REQUESTED** [review]
I love the tests, great job!  Thanks for reusing the existing parsing logic, rather than adding dependencies.
Some tiny comments below:

File: src/urls.ts (hunk 1 above)
Comment (jmisasa) [reply src/urls.ts:29]: `UrlObject` type has `query` declared as optional

File: src/MemoryRouter.test.tsx (hunk 2 above)
Comment (jmisasa) [reply src/MemoryRouter.test.tsx:450]: yes exactly, tested this manually and this seems to be the behaviour

**Review by scottrippey — APPROVED** [review]

**Pull request merged**